test/src/tests.js

```diff
@@ -83,29 +85,163 @@ describe('Rokt Forwarder', () => {
+
+        it('should call launcher.selectPlacements with all passed through options', async () => {
+            window.mParticle.forwarder.filters = {
+                userAttributesFilters: [],
+                filterUserAttributes: function (attributes) {
+                    return attributes;
+                },
+            };
+
+            await window.mParticle.forwarder.init(
+                {
+                    accountId: '123456',
+                },
+                reportService.cb,
+                true,
+                null,
+                {}
+            );
+
+            await window.mParticle.forwarder.selectPlacements({
+                identifier: 'test-placement',
+                attributes: {
+                    test: 'test',
+                },
+            });
 
-        await mParticle.forwarder.init(
-            {
-                accountId: '123456',
-            },
-            reportService.cb,
-            true
-        );
+            window.Rokt.selectPlacementsCalled.should.equal(true);
+            window.Rokt.selectPlacementsOptions.should.deepEqual({
+                identifier: 'test-placement',
+                attributes: {
+                    test: 'test',
+                },
+            });
+        });
+
+        it('should call launcher.selectPlacements with filtered user attributes', async () => {
+            window.mParticle.forwarder.filters = {
+                userAttributesFilters: [],
+                filterUserAttributes: function () {
+                    return {
+                        'user-attribute': 'user-attribute-value',
+                        'unfiltered-attribute': 'unfiltered-value',
+                    };
+                },
+            };
+
+            await window.mParticle.forwarder.init(
+                {
+                    accountId: '123456',
+                },
+                reportService.cb,
+                true,
+                null,
+                {}
+            );
+
+            await window.mParticle.forwarder.selectPlacements({
+                identifier: 'test-placement',
+                attributes: {
+                    'unfiltered-attribute': 'unfiltered-value',
+                    'filtered-attribute': 'filtered-value',
+                },
+            });
 
-        window.Rokt.accountId.should.equal('123456');
+            window.Rokt.selectPlacementsCalled.should.equal(true);
+            window.Rokt.selectPlacementsOptions.should.deepEqual({
+                identifier: 'test-placement',
+                attributes: {
+                    'user-attribute': 'user-attribute-value',
+                    'unfiltered-attribute': 'unfiltered-value',
+                },
+            });
+        });
+
+        it('should process user attributes filters that are passed in via the config and selectPlacements options', async () => {
+            window.mParticle.forwarder.filters = {
+                userAttributesFilters: [],
```

Comment:
the test says it will process the user attribute filters that are passed on the config, but there are no filters here, so the attributes end up being the same as what is passed in.

Reply:
Technically the filters are used by `filterUserAttributes` via the Core SDK, and in this case we're mocking that function. The true purpose of this test is to verify that `selectPlacements` is correctly using the results of `filterUserAttributes`. Hope this change helps.